pkg/engine/image.go

```diff
@@ -45,29 +61,117 @@ func (i *Image) Apply(ctx, conn context.Context, currentState, desiredState plum
 	return nil
 }
 
-func (i *Image) loadPodman(ctx, conn context.Context, url string) error {
-	klog.Infof("Loading image from %s", i.Url)
-	// Place the data into the placeholder file
-	data, err := http.Get(i.Url)
+func (i *Image) loadHTTPPodman(ctx, conn context.Context, url, target string) error {
+	imageName := (path.Base(url))
+	pathToLoad := "/opt/" + imageName
+	data, err := http.Get(url)
 	if err != nil {
-		klog.Error("Failed getting data from ", i.Url)
-		return err
+		// remove the image if it exists
+		if _, err := os.Stat(pathToLoad); err == nil {
```

Comment:
klog the err above, then, if err != nil here, return this err
then, don't need the err==nil, just the klog& flushImages below